src/createPackage/models/createPackageManager.ts

```diff
@@ -267,6 +262,24 @@ export class CreatePackageManager {
     return sanitized;
   }
 
+  private generateTileGroups(polygon: Polygon, footprint: Polygon | MultiPolygon, zoom: number): ITileRange[] | null {
+    const intersaction = intersect(polygon, footprint);
+    const tilesGroups: ITileRange[] = [];
+
+    if (intersaction === null) {
+      return null;
+    }
+
+    for (let i = 0; i <= zoom; i++) {
```

Comment:
you dont need the for, just encode with max zoom level and check the result GPKG

Reply:
Open an issue regarding the expected and actual behavior of the algorithm and i'll approve this

Reply:
done [](https://mapcolonies.atlassian.net/browse/MAPCO-2871)

Reply:
https://mapcolonies.atlassian.net/browse/MAPCO-2871